test/scenarios/withnetns.py

```diff
@@ -10,153 +10,91 @@
-)
-
-# test that netns-exec can not be executed by users that are not operator
-machine.fail("sudo -u clightning netns-exec nb-bitcoind ip a")
-
-# test that `systemctl status` can't leak credentials
-assert_matches(
-    "sudo -u electrs systemctl status clightning 2>&1 >/dev/null",
-    "Failed to dump process list for 'clightning.service', ignoring: Access denied",
-)
-machine.succeed("grep -Fq hidepid=2 /proc/mounts")
-
-### Additional tests
-
-# Current time in µs
-pre_restart = succeed("date +%s.%6N").rstrip()
-
-# Sanity-check system by restarting all services
-succeed("systemctl restart bitcoind clightning spark-wallet lightning-charge nanopos liquidd")
-
-# Now that the bitcoind restart triggered a banlist import restart, check that
-# re-importing already banned addresses works
-machine.wait_until_succeeds(
-    log_has_string(f"bitcoind-import-banlist --since=@{pre_restart}", "Importing node banlist")
-)
-assert_no_failure("bitcoind-import-banlist")
-
-### Test lnd
-
-succeed("systemctl stop nanopos lightning-charge spark-wallet clightning")
-succeed("systemctl start lnd")
-assert_matches("su operator -c 'lncli getinfo' | jq", '"version"')
-assert_no_failure("lnd")
-
-### Test loopd
-
-succeed("systemctl start lightning-loop")
-assert_matches("su operator -c 'loop --version'", "version")
-# Check that lightning-loop fails with the right error, making sure
-# lightning-loop can connect to lnd
-machine.wait_until_succeeds(
-    log_has_string("lightning-loop", "chain notifier RPC isstill in the process of starting")
-)
+
+def electrs():
+    machine.wait_until_succeeds(
+        "ip netns exec nb-electrs nc -z localhost 4224"
+    )  # prometeus metrics provider
+
+
+def spark_wallet():
+    machine.wait_until_succeeds("ip netns exec nb-spark-wallet nc -z %s 9737" % sparkwallet_ip)
+    spark_auth = re.search("login=(.*)", succeed("cat /secrets/spark-wallet-login"))[1]
+    assert_matches(
+        f"ip netns exec nb-spark-wallet curl -s {spark_auth}@%s:9737" % sparkwallet_ip, "Spark"
+    )
+
+
+def lightning_charge():
+    machine.wait_until_succeeds("ip netns exec nb-nanopos nc -z %s 9112" % lightningcharge_ip)
+    charge_auth = re.search("API_TOKEN=(.*)", succeed("cat /secrets/lightning-charge-env"))[1]
+    assert_matches(
+        f"ip netns exec nb-nanopos curl -s api-token:{charge_auth}@%s:9112/info | jq"
+        % lightningcharge_ip,
+        '"id"',
+    )
+
+
+def nanopos():
+    machine.wait_until_succeeds("ip netns exec nb-lightning-charge nc -z %s 9116" % nanopos_ip)
+    assert_matches("ip netns exec nb-lightning-charge curl %s:9116" % nanopos_ip, "tshirt")
+
+
+def web_index():
+    machine.wait_until_succeeds("ip netns exec nb-nginx nc -z localhost 80")
+    assert_matches("ip netns exec nb-nginx curl localhost", "nix-bitcoin")
+    assert_matches("ip netns exec nb-nginx curl -L localhost/store", "tshirt")
+
+
+def post_clightning():
```

Comment:
I think this should be called `netns_isolation` since all its tests concern Network Namespace Isolation

Reply:
if this is called netns isolation it's not clear if it's executed in the clightning or in the lnd world. Makes more sense to me to name the keys in extra_tests according to the location it will be executed.

Reply:
What about `final`?